jccli/jc_api_v1.py

```diff
@@ -113,3 +114,32 @@ def get_user_id(self, username):
         for user in users:
             if user.username == username:
                 return user.id
+
+    def get_user(self, username):
+        """
+        Get detail view of a user object.
+        :param user_id:
+        :return: user properties dict
+        """
+        # TODO: Does it make sense to make two API calls (one to get the ID using the username, then another to get the
```

Comment:
Conceivably the CLI could cache username-id pairs locally, I suppose.

Other than that, all of the JCAPI-Python calls seem to rely on "id", not "username", which is not what our user is going to be working with.

Reply:
One thing we definitely should do is to make a filtered list request, e.g. something like`systemusers_list(filter='username:%s' % (username,))`, instead of what we are doing currently which is request a list of _all_ users then iterate through it locally until we get the user with the right username.

The issue with that is that I still don't fully understand how the `filter` parameter works (see [issue #46](https://github.com/TheJumpCloud/jcapi-python/issues/46)).

Reply:
> Conceivably the CLI could cache username-id pairs locally, I suppose.

That's a good idea, let's explore as a refactor later on.

> Other than that, all of the JCAPI-Python calls seem to rely on "id", not "username", which is not what our user is going to be working with.

yep, you are correct.  I implemented with `username` because it was easier than id.  Maybe we should leave this portion as-is and do `id` for the other commands to get to parity with the powershell module?  then we can explore username afterwards?

Reply:
Now that I think about it, since this is the get-user function, it could just return the user object directly from a call to the list view (I think), so there is indeed no need for two sequential API calls in this particular case; I think I'll make that change in this PR.

However, this conversation does still apply to other functions in the CLI (in particular, set-user & delete-user)